service_capacity_modeling/models/org/netflix/ddb.py

```diff
@@ -0,0 +1,584 @@
+    assert transactional_read_percent >= 0.0
+    if (
+        desires.query_pattern.access_consistency.same_region.target_consistency
+        == AccessConsistency.read_your_writes
+    ):
+        eventual_consistency_percent = 0.0
+    strong_consistency_percent = (
+        1 - eventual_consistency_percent - transactional_read_percent
+    )
+    assert strong_consistency_percent >= 0.0
+    mean_item_size = _mean_item_size_bytes(desires, extra_model_arguments)
+    rounded_rcus_per_item = math.ceil(max(1.0, mean_item_size / (4 * 1024)))
+    estimated_read_per_second = desires.query_pattern.estimated_read_per_second.mid
+    baseline_rcus_eventually_consistent = (
+        estimated_read_per_second
+        * eventual_consistency_percent
+        * 0.5
+        * rounded_rcus_per_item
+    )
+    baseline_rcus_strong_consistent = (
+        estimated_read_per_second
+        * strong_consistency_percent
+        * 1
+        * rounded_rcus_per_item
+    )
+    baseline_rcus_transact = (
+        estimated_read_per_second
+        * transactional_read_percent
+        * 2
+        * rounded_rcus_per_item
+    )
+
+    total_baseline_rcus = math.ceil(
+        baseline_rcus_eventually_consistent
+        + baseline_rcus_strong_consistent
+        + baseline_rcus_transact
+    )
+
+    # 8760 hours in a year
+    total_baseline_rcus_hours = 8760 * total_baseline_rcus
+    dynamo_service_standard = _get_dynamo_standard(context, extra_model_arguments)
+    total_annual_cost_rcu_hour = round(
+        (total_baseline_rcus_hours * dynamo_service_standard.annual_cost_per_read_io), 2
+    )
+    return _ReadPlan(
+        read_capacity_units=total_baseline_rcus_hours,
+        total_annual_read_cost=total_annual_cost_rcu_hour,
+    )
+
+
+def _plan_storage(
+    context: RegionContext,
+    desires: CapacityDesires,
+    extra_model_arguments: Dict[str, Any],
+) -> _DataStoragePlan:
+    dynamo_service_standard = _get_dynamo_standard(context, extra_model_arguments)
+    annual_storage_cost = round(
+        (
+            math.ceil(desires.data_shape.estimated_state_size_gib.mid)
+            * dynamo_service_standard.annual_cost_per_gib[0][1]
+        ),
+        2,
+    )
+    return _DataStoragePlan(
+        total_data_storage_gib=round(
+            desires.data_shape.estimated_state_size_gib.mid, 2
+        ),
+        total_annual_data_storage_cost=annual_storage_cost,
+    )
+
+
+def _plan_data_transfer(
+    context: RegionContext,
+    desires: CapacityDesires,
+    extra_model_arguments: Dict[str, Any],
+) -> _DataTransferPlan:
+    number_of_regions = _get_num_regions(extra_model_arguments)
+    if not number_of_regions > 1:
+        return _DataTransferPlan(
+            total_data_transfer_gib=0, total_annual_data_transfer_cost=0
+        )
+    mean_item_size_bytes = _mean_item_size_bytes(desires, extra_model_arguments)
+    writes_per_second = desires.query_pattern.estimated_write_per_second.mid
+    # 31,536,000 seconds in a year
```

Comment:
nit: I often like to define constants at the top like this to make these multiplications easier to follow and check the units on

```
# Approximate seconds in a year
SECONDS_IN_YEAR= (365 * 24 * 60 * 60) # 31536000
# True time in a year if we want to be pedantic https://en.wikipedia.org/wiki/Year
SECONDS_IN_YEAR= 31556926

# 1GiB in bytes
GIB_IN_BYTES = 1024 * 1024 * 1024
```